### ISSUE TITLE
Background is color 0 and not background color on nvim nightly

### ISSUE THREAD

**Issue by Kagia001** [joshdick/onedark.vim#247]
- [x] I have carefully read and followed the [Installation section of the README](https://github.com/joshdick/onedark.vim#installation)
- [x] I have searched through the [closed issues](https://github.com/joshdick/onedark.vim/issues?q=is%3Aissue+is%3Aclosed) to see if there is an existing solution for my issue

### Terminal Emulator Info

ST 0.8.4
XTerm(365)
Alacritty 0.7.1 (a852eb85)
QTermWidget (IDK how to get the version, probably whatever is newest)

### Output From `vim --version`

```
NVIM v0.5.0-dev+1079-g8becdec39
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/clang -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/karl/.cache/paru/clone/neovim-nightly-git/src/build/config -I/home/karl/.cache/paru/clone/neovim-nightly-git/src/neovim-nightly-git/src -I/usr/include -I/home/karl/.cache/paru/clone/neovim-nightly-git/src/build/src/nvim/auto -I/home/karl/.cache/paru/clone/neovim-nightly-git/src/build/include
Compiled by karl

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info


```

### Issue Description

When using 16 color mode, the background is color 0 and not bg (see screenshot). Same effects as issue #202, although it probably has a different cause. This only happens on nvim nightly, the theme works normally on nvim, so there probably is an incomatibility there. I have tried a few different terminals, the issue happens with all of them. For my case I can just use true color instead, but I feel like you should be aware of this problem.
![screenshot](https://user-images.githubusercontent.com/48362551/107343929-24e32e00-6ac2-11eb-9594-159d6e6f5f34.jpg)

minimal config with issue: (lua)
```
vim.g.onedark_termcolors = 16
vim.api.nvim_command('syntax on')
vim.api.nvim_command('colorscheme onedark')

-----package manager------------
vim.cmd 'packadd paq-nvim'
local paq = require'paq-nvim'.paq
paq{'savq/paq-nvim', opt=true}

paq 'joshdick/onedark.vim'
```

**Comment by joshdick** [comment]
@Kagia001 Thanks for the report. I have not been closely following the nightly builds of Neovim. If this is only happening in nightly, it's not clear whether the cause is due to an intentional and documented change on Neovim's part, or whether it's due to a regression. Is it worth reporting this to Neovim as well?

**Comment by joshdick** [comment]
#271 that was recently merged, which has been tested with Neovim 0.5, explicitly uses the terminal background color for the background in 16-color mode, instead of ANSI color 0 (see the PR description for details.)

Closing due to inactivity, but feel free to keep the conversation going if needed!
